shim/opentelemetry-opencensus-shim/src/opentelemetry/shim/opencensus/_shim_tracer.py

```diff
@@ -67,20 +124,27 @@ def start_span(self, name="span"):
         # equivalent to the below. This can cause context to leak but is equivalent.
         # pylint: disable=protected-access
         shim_span._self_token = context.attach(ctx)
+        # Also set it in OC's context, equivalent to
+        # https://github.com/census-instrumentation/opencensus-python/blob/2e08df591b507612b3968be8c2538dedbf8fab37/opencensus/trace/tracers/context_tracer.py#L94
+        execution_context.set_current_span(shim_span)
         return shim_span
 
     def end_span(self):
-        """Finishes the current span in the context and pops restores the context from before
-        the span was started.
+        """Finishes the current span in the context and restores the context from before the
+        span was started.
         """
         span = self.current_span()
         if not span:
             _logger.warning("No active span, cannot do end_span.")
             return
 
         span.finish()
+
         # pylint: disable=protected-access
         context.detach(span._self_token)
+        # Also reset the OC execution_context, equivalent to
+        # https://github.com/census-instrumentation/opencensus-python/blob/2e08df591b507612b3968be8c2538dedbf8fab37/opencensus/trace/tracers/context_tracer.py#L114-L117
+        execution_context.set_current_span(self.current_span())
 
```

Comment:
`self.current_span()` is updated to the previous span once `context.detach(span._self_token)` is called right?

Reply:
Yes that's right.